packages/aquery/src/utils/form/datePicker.js

```diff
@@ -3,39 +3,72 @@ import randomId from '../randomId';
 const FORMATS = { FORMAT_US: 'MM-DD-YYYY', FORMAT_WORLD: 'DD-MM-YYYY' };
 
 const addDatePicker = (date, $) => {
-  if (date) {
-    const input = date.input;
-    if (!input.getAttribute('name')) {
-      input.setAttribute('name', randomId());
-    }
-    const datePicker = document.createElement('amp-date-picker');
-    datePicker.setAttribute(
-      'input-selector',
-      `[name=${input.getAttribute('name')}]`,
-    );
-    if (date.mode && date.mode === 'static') {
-    } else {
-      datePicker.setAttribute('mode', 'overlay');
-      datePicker.setAttribute('layout', 'container');
-    }
-    const format = resolveFormat(date.format);
-    datePicker.setAttribute('format', format);
-    if (date.required) {
-      input.removeAttribute('readonly');
-      input.setAttribute('required', '');
-      datePicker.setAttribute('required', '');
-      console.log('addDatePicker ', datePicker);
+    if (date) {
+        const mode = date.mode;
+        const input = date.input;
+        if (!input.getAttribute('name')) {
+            input.setAttribute('name', randomId());
+        }
+        const datePicker = document.createElement('amp-date-picker');
+        datePicker.setAttribute(
+            'input-selector',
+            `[name=${input.getAttribute('name')}]`,
+        );
+        if (date.mode && date.mode === 'static') {} else if (mode === "lightbox") {
+            var lightboxID = `lb${input.getAttribute('name')}`;
+            datePicker.setAttribute('mode', 'static');
+            datePicker.setAttribute('layout', 'fixed-height');
+            datePicker.setAttribute('height', '360');
+            input.setAttribute('on', `tap:${lightboxID}.open`);
+            input.setAttribute('tabindex', '0');
+            input.setAttribute('role', 'textbox');
+        } else {
+            datePicker.setAttribute('mode', 'overlay');
+            datePicker.setAttribute('layout', 'container');
+        }
+        const format = resolveFormat(date.format);
+        datePicker.setAttribute('format', format);
+
+        if (date.required) {
+            input.removeAttribute('readonly');
+            input.setAttribute('required', '');
+            datePicker.setAttribute('required', '');
+            console.log('addDatePicker', datePicker);
+        }
+        if (mode === "lightbox") {
+            const lightbox = document.createElement('amp-lightbox');
```

Comment:
From ln 38 to 63 should all go under the if else above. No need to check if light box again

Reply:
But there are  changes that are made to the datepicker before line 38. If I'll put this block before these changes (line 30), these changes would be missed. 
2nd reason is that there is an 'else' to this 'if'

Reply:
I don't understand your point.
I am looking at the code, and there is no reason to check if mode === "lightbox" twice

If you don't manage to do that, ping me tomorrow and we can do it together

Reply:
OK I managed to do it.